In [2]:
import pandas as pd

# Работа с Pandas

1. Загрузите ваш DataFrame в переменную df и выведете на экран первые пять строк 

In [3]:
df = pd.read_csv("C:/Users/Z/PycharmProjects/googleplaystore.csv")
#Переименовываем столбцы
df.rename(columns={'App':'app','Category':'category','Rating':'rating', 'Reviews':'reviews', 'Size':'size', 'Installs':'installs', 'Type':'type', 'Price':'price', 'Content Rating':'content_rating', 'Genres':'genres', 'Last Updated':'last_updated', 'Current Ver':'current_ver', 'Android Ver':'android_ver'},inplace=True)
#Выводим первые пять строк
df.head()

app        category  rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  reviews  size     installs  type  price content_rating  \
0     159   19M      10,000+  Free    0.0       Everyone   
1     967   14M     500,000+  Free    0.0       Everyone   
2   87510  8.7M   5,000,000+  Free    0.0       Everyone   
3  215644   25M  50,000,000+  Free    0.0           Teen   
4     967  2.8M     100,000+  Free    0.0       Everyone   

                      genres      last_updated         current_ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    android_ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

2. Отобразите количество строк и столбцов в DataFrame

In [4]:
df.shape

(10841, 13)

3. Выведете основную информацию о DataFrame методом info(). Проанализируйте есть ли пропуски в столбцах. Напишите вывод о том, какие столбцы содержат пропуски

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   app             10841 non-null  object 
 1   category        10841 non-null  object 
 2   rating          9367 non-null   float64
 3   reviews         10841 non-null  object 
 4   size            10841 non-null  object 
 5   installs        10841 non-null  object 
 6   type            10840 non-null  object 
 7   price           10841 non-null  float64
 8   content_rating  10840 non-null  object 
 9   genres          10841 non-null  object 
 10  last_updated    10841 non-null  object 
 11  current_ver     10833 non-null  object 
 12  android_ver     10838 non-null  object 
dtypes: float64(2), object(11)
memory usage: 1.1+ MB


<b>Вывод:</b>

В представленном DataFrame 13 столбцов, из которых:
* 11 имеют тип `object`
* 2 имеют тип `float64`
* всего в датафрейме 10841 строка
* в столбцах 'Rating', 'Type', 'Content Rating', 'Current Ver' и 'Android Ver' есть пропуски (`NaN`)

4. Удалите или замените пропуски известными вам методами. (Подсказка: `dropna()` или `fillna()`, выбор того или иного метода должен быть аргументирован).

<b>Аргументация выбора методов:</b>

1. Рейтинг приложения величина сугубо индивидуальная, заменять усредненным значением считаю нецелесообразным
2. Оставшуюся незначительную часть пустых данных можно удалить

In [6]:
df.rating.fillna(0, inplace=True)
df.dropna(inplace=True)
#Приводим индексы в порядок
df.reset_index(drop=True)

app             category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10824                                   Sya9a Maroc - FR               FAMILY   
10825                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10826                             Parkinson Exercices FR              MEDICAL   
10827                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10828      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       rating reviews                size     installs  type  price  \
0         4.1     159                 19M      10,000+  Free    0.0   
1         3.9     967                 14M     500,000+  Free    0.0   
2         4.7   87510                8.7M   5,000,000+  Free    0.0   
3         4.5  215644                 25M  50,000,000+  Free    0.0   
4         4.3     967                2.8M     100,000+  Free    0.0   
...       ...     ...                 ...          ...   ...    ...   
10824     4.5      38                 53M       5,000+  Free    0.0   
10825     5.0       4                3.6M         100+  Free    0.0   
10826     0.0       3                9.5M       1,000+  Free    0.0   
10827     4.5     114  Varies with device       1,000+  Free    0.0   
10828     4.5  398307                 19M  10,000,000+  Free    0.0   

      content_rating                     genres      last_updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10824       Everyone                  Education     July 25, 2017   
10825       Everyone                  Education      July 6, 2018   
10826       Everyone                    Medical  January 20, 2017   
10827     Mature 17+          Books & Reference  January 19, 2015   
10828       Everyone                  Lifestyle     July 25, 2018   

              current_ver         android_ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10824                1.48          4.1 and up  
10825                 1.0          4.1 and up  
10826                 1.0          2.2 and up  
10827  Varies with device  Varies with device  
10828  Varies with device  Varies with device  

[10829 rows x 13 columns]

5. Проверьте нет ли пропусков после произведенной обработки

In [16]:
df.isna().sum()

app               0
category          0
rating            0
reviews           0
size              0
installs          0
type              0
price             0
content_rating    0
genres            0
last_updated      0
current_ver       0
android_ver       0
dtype: int64

6. Выведите на экран минимум и максимум из столбца `Price`:

In [8]:
print(df.price.max())
print(df.price.min())

400.0
0.0


7. Выведите на экран медиану и среднее арифметическое столбцов:
 - Rating
 - Reviews

In [9]:
print(df.rating.mean())
print(df.rating.median())
#
print(df.reviews.astype(float).mean())
print(df.reviews.median())

3.6231969710961307
4.2
444601.76673746423
2100.0


8. Выведите на экран все уникальные значения категориального столбца Genres

In [10]:
df.genres.unique()

array(['Art & Design', 'Art & Design;Pretend Play',
       'Art & Design;Creativity', 'Art & Design;Action & Adventure',
       'Auto & Vehicles', 'Beauty', 'Books & Reference', 'Business',
       'Comics', 'Comics;Creativity', 'Communication', 'Dating',
       'Education;Education', 'Education', 'Education;Creativity',
       'Education;Music & Video', 'Education;Action & Adventure',
       'Education;Pretend Play', 'Education;Brain Games', 'Entertainment',
       'Entertainment;Music & Video', 'Entertainment;Brain Games',
       'Entertainment;Creativity', 'Events', 'Finance', 'Food & Drink',
       'Health & Fitness', 'House & Home', 'Libraries & Demo',
       'Lifestyle', 'Lifestyle;Pretend Play',
       'Adventure;Action & Adventure', 'Arcade', 'Casual', 'Card',
       'Casual;Pretend Play', 'Action', 'Strategy', 'Puzzle', 'Sports',
       'Music', 'Word', 'Racing', 'Casual;Creativity',
       'Casual;Action & Adventure', 'Simulation', 'Adventure', 'Board',
       'Trivia', 'Role 

9. Сгруппируйте данные по столбцу `Genres` и посчитайте для каждого жанра средний и медианный рейтинг. Представьте результат в виде нового DataFrame,который будет сохранен в переменную `grouped_df`, где одна колонка будет содержать названия жарнов, а две другие буду содержать в себе средний и медианный рейтинги. Напишите краткий вывод.

In [11]:
grouped_df = df.groupby(by='genres', as_index=False).agg(rating_mean = ('rating', "mean"), rating_median = ('rating', "median"))
grouped_df

genres  rating_mean  rating_median
0                                   Action     4.203288            4.3
1                Action;Action & Adventure     4.311765            4.3
2                                Adventure     4.069333            4.3
3             Adventure;Action & Adventure     4.423077            4.5
4                    Adventure;Brain Games     4.600000            4.6
..                                     ...          ...            ...
114                Video Players & Editors     3.711561            4.2
115     Video Players & Editors;Creativity     4.100000            4.1
116  Video Players & Editors;Music & Video     4.000000            4.0
117                                Weather     3.881707            4.2
118                                   Word     4.258621            4.4

[119 rows x 3 columns]

<b>Выводы:</b>

1. Всего в датасете содержится 119 различных жанров

10. Из получившегося датафрейма выведите на экран только те жанры, чей медианный рейтинг больше 4.5. Напишите краткий вывод.

In [12]:
msk = grouped_df.rating_median > 4.5
grouped_df[msk]

genres  rating_mean  rating_median
4          Adventure;Brain Games     4.600000            4.6
11       Art & Design;Creativity     4.400000            4.7
18            Board;Pretend Play     4.800000            4.8
35             Comics;Creativity     4.800000            4.8
55      Entertainment;Creativity     4.533333            4.6
64    Health & Fitness;Education     4.700000            4.7
74           Music;Music & Video     4.533333            4.6
87              Puzzle;Education     4.600000            4.6
105  Strategy;Action & Adventure     4.600000            4.6

<b>Выводы:</b>

1. Самый высокий рейтинг имеют приложения из жанров 'Board;Pretend Play' и 'Comics;Creativity'.
2. Рейтинг жанра'Art & Design;Creativity' имеет не симметричное распределение с левосторонней скошенностью.

11. Выведете кол-во дубликатов в основном DataFrame (df) и удалите их 

In [13]:
#Считаем дубликаты
x = len(df) - len(df.drop_duplicates())
#Удаляем дубликаты
df.drop_duplicates(inplace=True)
x

483

12. Посчитатайте какое количество игр содержится в каждом жанре (value_counts). Выведите на экран топ-10 игр по количеству в жанре. Напишите краткий вывод.

In [14]:
df.genres.value_counts().head(10)

genres
Tools              840
Entertainment      587
Education          526
Business           427
Medical            408
Productivity       407
Personalization    386
Lifestyle          372
Communication      366
Sports             364
Name: count, dtype: int64

<b>Выводы:</b>

1. Самым популярным является жанр 'Tools'